packages/grid/x-data-grid/src/utils/styled.ts

```diff
@@ -0,0 +1,4 @@
+import { createStyled } from '@mui/system';
+
+// hardcode theme id to avoid importing from `@mui/material`
+export const styled = createStyled({ themeId: '$$material' });
```

Comment:
Side note: should we export it publicly? Which `styled` should devs use when e.g. creating a custom slot for the Data Grid?

Reply:
They could use the one from `@mui/material/styles`, no? IIANM, the need to de-isolate `material/styles` is more for `internal` use.
We could also expose this one and ask them to use it but it might be difficult to effectively communicate this to the users. Wdyt?

Reply:
Would it make more sense to switch from `@mui/system` to use `styled` from `@mui/material/styles` for now?

I recall that the use of `@mui/system` is to support Joy in the future but since the core team is focusing on v6, we could revisit and provide a better solution in the future.

Reply:
I am going to work on bundling the grid without MaterialUI this quarter though. Which implies we remove any `@mui/material` import from the core of the grid's codebase.

Reply:
> Would it make more sense to switch from @mui/system to use styled from @mui/material/styles for now?

Thanks for pitching in @siriwatknp 

I would agree with @romgrk and possibly avoid reverting to an import from `@mui/material`. IMO, this workaround could be a nice compromise for the time being.

Or do you see a problem with this other than this duplicated hardcoded import?